### PR TITLE
Feature eks

### LINKED ISSUE #2: EKS Arch
This issue is for the PR to be raised for the overall update to the way we want our EKS architecture.
Considering the same, we would be making this a major change.

github_eks

### PR DESCRIPTION
**Pull request by tprakash17**
Closes #2 

This PR has the entire follow up architectural changes we want to bring in.

### CODE REVIEW

File: frankfurt-eks-vpc.tf [hunk 1]
```diff
@@ -0,0 +1,114 @@
+#
```
Comment (ntantri) [inline frankfurt-eks-vpc.tf]: This file name should have been with `frankfurt`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,115 @@
...
+```
+
+#### Apply changes
+```
+$ terraform apply
+```
+
+#### Configure kubectl
+```
+$ aws eks --region eu-central-1 update-kubeconfig --name <CLUSTER-NAME>
```
Comment (ntantri) [inline README.md]: Can you please change this to:

`aws eks --region <AWS-REGION> update-kubeconfig --name <CLUSTER-NAME>`

**Review by ntantri — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (tprakash17) [reply README.md]: yo

File: frankfurt-eks-vpc.tf (hunk 1 above)
Comment (tprakash17) [reply frankfurt-eks-vpc.tf]: renamed file to eks-vpc.tf

File: README.md (hunk 2 above)
Comment (tprakash17) [reply README.md]: done

**Review by ntantri — CHANGES_REQUESTED** [review]
1. Can you ensure all the terraform files have proper comments which indicate the files usage and things for others to know the importance of those files? Readme would help them get the structure identified for installation and deployment, but comments would help them link the scenarios.
2. Ensure to add proper folder structures for the files that are in reference here. Meaning, see if you can club files and add under meaningful folders. For example, all the Auth related can go in one folder named `auth_setup` and there we can have the files like `iam.tf` and so on in that folder. Please add that logical segregation for the files.
3. As well I see the `aws-auth.yaml` is empty, can you add a comment there, saying this should have the credentials to be pasted from the output of the configMap and how it looks like?

**Comment by tprakash17** [comment]
@tan31989 

1st comment - Yes, will address this.

2nd comment - All of these `.tf` files should be with in current directory where `terraform plan or apply` runs can't be in different folders. There is concept of module but is applied for higher level config such EKS cluster itself not for small small files such as auth. Let me know if you want more clarification.

3rd comment - will address.

**Comment by tprakash17** [comment]
@tan31989 updated description for the resources within files and also for `aws-auth.yaml`.

**Review by ntantri — APPROVED** [review]

**Pull request merged**